barman/cloud_providers/google_cloud_storage.py

```diff
@@ -575,102 +582,169 @@ def delete_snapshot_backup(self, backup_info):
+            attached_volumes[disk_name] = GcpVolumeMetadata(
+                attachment_metadata,
+                disk_metadata,
+            )
+        # Check all requested disks were found and complain if necessary
+        if disks is not None:
+            unattached_disks = []
+            for disk_name in disks:
+                if disk_name not in attached_volumes:
+                    # Verify the disk definitely exists by fetching the metadata
+                    self._get_disk_metadata(disk_name)
+                    # Append to list of unattached disks
+                    unattached_disks.append(disk_name)
+            if len(unattached_disks) > 0:
                 raise SnapshotBackupException(
-                    "Disk %s appears to be attached with name %s as devices %s and %s"
-                    % (
-                        attached_disk.source,
-                        disk_name,
-                        full_device_name,
-                        attached_devices[disk_name],
-                    )
+                    "Disks not attached to instance %s: %s"
+                    % (instance_name, ", ".join(unattached_disks))
                 )
-            attached_devices[disk_name] = full_device_name
-
-        return attached_devices
+        return attached_volumes
 
-    def get_attached_snapshots(self, instance_name, zone):
+    def instance_exists(self, instance_name):
         """
-        Returns the snapshots which are sources for disks attached to instance.
-
-        Queries the instance metadata to determine which disks are attached and
-        then queries the disk metadata for each disk to determine whether it was
-        cloned from a snapshot. If it was cloned then the snapshot is added to the
-        dict which is returned once all attached devices have been checked.
+        Determine whether the named instance exists.
 
         :param str instance_name: The name of the VM instance to which the disks
             to be backed up are attached.
-        :param str zone: The zone in which the snapshot disks and instance reside.
-        :rtype: dict[str,str]
-        :return: A dict where the key is the snapshot name and the value is the
-            device path for the source disk for that snapshot on the specified
-            instance.
-        """
-        attached_devices = self.get_attached_devices(instance_name, zone)
-        attached_snapshots = {}
-        for disk_name, device_name in attached_devices.items():
-            disk_metadata = self._get_disk_metadata(disk_name, zone)
+        :rtype: bool
+        :return: True if the named instance exists, False otherwise.
+        """
+        try:
+            self.instances_client.get(
+                instance=instance_name,
+                zone=self.zone,
+                project=self.project,
+            )
+        except NotFound:
+            return False
+        return True
+
+
+class GcpVolumeMetadata(VolumeMetadata):
```

Comment:
I know it deviates from PEP-8 but personally I find treating acronyms like regular words in CamelCase much easier to read, particularly when when two acronyms end up next to each other in a name (not that that's the case here, but it might happen at some point).

Given Barman already has class names like `WalArchiverQueue`, `CsvOption` and `SshCommandException` I think there's also a case to be made for "local style overrides PEP-8 style". That said, I can see at least one counter-example: `WALFileException` and there are likely others too...